### DISCUSSION TITLE
Tweaking Coordinator Refresh - Is it safe?

### DISCUSSION THREAD

**Discussion by RobinXe** [BottlecapDave/HomeAssistant-OctopusEnergy#634] [General]
BLUF: I've checked the code, and I'm fairly sure it's fine but wanted to throw the question out there in case I've missed an instance:

If I change the constant COORDINATOR_REFRESH_IN_SECONDS to something lower than 60, will the result merely be (slightly) higher load on my HA server, or do I risk spamming the API?

Context: The integration works amazingly, and triggers my system to start/stop discharging in <60 s of a Saving Session starting/ending. As a profligate minmaxer however, my CDO is tweaked by not getting it (closer to) bang on time. It would appear that lowering the COORDINATOR_REFRESH_IN_SECONDS constant would achieve this, by evaluating whether now() is inside a Saving Session period with greater frequency, but I'm concerned that this value may be used, somewhere I've missed, for Octopus API polling, and I am seeking reassurance from the community (particularly Dave) that doing so would be fine.

Cheers!

**Comment by BottlecapDave** [comment]
In the current itteration, you will run the risk of spamming the API as it will be called twice (or more) when it hits the hourly/half hourly mark for certain services (e.g. rates). However, this shouldn't occur if you do it on the current beta as it tracks it's own update schedule (I have it on my list to try and lower this and entity update values). It will obviously also apply across all entities and coordinators so some of the more expensive processing may be processed more and result in degradation.

 I also won't be able to assist with any issues for people that are modifying the codebase.

**Comment by RobinXe** [comment]
Understood, thanks Dave!

I appreciate that people run HA on a variety of hardware, with varying amounts of other tasks to run, so maybe a 'nice-to-have' would be a setting in the configuration menu for the frequency with which data 'cached' within properties is processed/examined, for a potentially faster response to triggers like saving sessions or rate drops etc.? Happy to raise this as an 'enhancement' request, if you think it's worthwhile.

**Reply by BottlecapDave** [reply]
I'd personally rather keep this type of configuration out, as it can be dangerous (more and more less technical people are using HA and the integration) and goes against what other integrations do. If people want to modify it themselves, they're obv free to do changes on their own forks.

As mentioned, it's on my list to see if this frequency can be increased (I run on a raspberry pi 3&4 which is prob the most likely and worst hardware) but unfortunately I need to balance it with the API requests which can be slow. There's also entity scan intervals to think about which apply to entities that don't use coordinator data (less of them). I'm personally not sure how HA reacts if the coordinator runs longer than the refresh period - I think it raises warnings which people would then run bugs against

**Reply by RobinXe** [reply]
Understood! I suspect you're right!
